clients/include/testing_axpy_strided_batched.hpp

```diff
@@ -0,0 +1,258 @@
+            //
+            // Transfer host to device
+            //
+            CHECK_HIP_ERROR(dy.transfer_from(hy));
+            CHECK_HIP_ERROR(dalpha.transfer_from(halpha));
+            //
+            // Call routine.
+            //
+            CHECK_ROCBLAS_ERROR(rocblas_axpy_strided_batched<T>(
+                handle, N, dalpha, dx, incx, stridex, dy, incy, stridey, batch_count));
+
+            //
+            // Transfer from device to host.
+            //
+            CHECK_HIP_ERROR(hy2.transfer_from(dy));
+
+            //
+            // CPU BLAS
+            //
+            {
+                cpu_time_used = get_time_us();
+
+                //
+                // Compute the host solution.
+                //
+                for(rocblas_int batch_index = 0; batch_index < batch_count; ++batch_index)
+                {
+                    cblas_axpy<T>(N, h_alpha, hx[batch_index], incx, hy[batch_index], incy);
+                }
+                cpu_time_used = get_time_us() - cpu_time_used;
+                cblas_gflops  = axpy_gflop_count<T>(N * batch_count) / cpu_time_used * 1e6;
+            }
+
+            //
+            // Compare with with the solution.
+            //
+            if(arg.unit_check)
+            {
+
+                unit_check_general<T>(1, N, batch_count, abs_incy, stridey, hy, hy1);
+
+                unit_check_general<T>(1, N, batch_count, abs_incy, stridey, hy, hy2);
+            }
+
+            if(arg.norm_check)
+            {
+                rocblas_error_1
+                    = norm_check_general<T>('I', 1, N, abs_incy, stridey, batch_count, hy, hy1);
+                rocblas_error_2
+                    = norm_check_general<T>('I', 1, N, abs_incy, stridey, batch_count, hy, hy2);
+            }
+        }
+    }
+
+    if(arg.timing)
+    {
+        int number_cold_calls = 2;
+        int number_hot_calls  = 100;
+        CHECK_ROCBLAS_ERROR(rocblas_set_pointer_mode(handle, rocblas_pointer_mode_host));
+
+        //
+        // Transfer from host to device.
+        //
+        CHECK_HIP_ERROR(dy.transfer_from(hy));
+
+        //
+        // Cold.
+        //
+        for(int iter = 0; iter < number_cold_calls; iter++)
+        {
+            rocblas_axpy_strided_batched<T>(
+                handle, N, &h_alpha, dx, incx, stridex, dy, incy, stridey, batch_count);
+        }
+
+        //
+        // Transfer from host to device.
+        //
+        CHECK_HIP_ERROR(dy.transfer_from(hy));
+
+        gpu_time_used = get_time_us(); // in microseconds
+        for(int iter = 0; iter < number_hot_calls; iter++)
+        {
+            rocblas_axpy_strided_batched<T>(
+                handle, N, &h_alpha, dx, incx, stridex, dy, incy, stridey, batch_count);
+        }
+        gpu_time_used     = (get_time_us() - gpu_time_used) / number_hot_calls;
+        rocblas_gflops    = axpy_gflop_count<T>(N * batch_count) / gpu_time_used * 1e6 * 1;
```

Comment:
* 1

Reply:
?

Reply:
You tell me what the *1 at the end of the line is supposed to indicate  :-)

Reply:
:)